modules/services/comodoro.nix

```
@@ -0,0 +1,72 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.comodoro;
+
+  args = with cfg.settings; {
+    inherit preset;
+    protocols = if lib.isList protocols then
+      lib.concatStringsSep " " protocols
+    else
+      protocols;
+  };
+
+in {
+  meta.maintainers = with lib.hm.maintainers; [ soywod ];
+
+  options.services.comodoro = {
+    enable = lib.mkEnableOption "Comodoro server";
+
+    package = lib.mkPackageOption pkgs "comodoro" { };
+
+    environment = lib.mkOption {
+      type = with lib.types; attrsOf str;
+      default = { };
+      example = lib.literalExpression ''
+        {
+          "PASSWORD_STORE_DIR" = "~/.password-store";
+        }
+      '';
+      description = ''
+        Extra environment variables to be exported in the service.
+      '';
+    };
+
+    settings = {
+      preset = lib.mkOption {
```

I don't think  `settings` is necessary here. It is not describing an RFC-42 style configuration like in the programs module.

```suggestion
    preset = lib.mkOption {
```